### ISSUE TITLE
Rule proposal: Template tag spacing

### ISSUE THREAD

**Issue by phaux** [eslint/eslint#7631]
**Please describe what the rule should do:** Similar to func-call-spacing – it would enforce spacing between template tag and template literal.

**What category of rule is this? (place an "X" next to just one item)**

- [X] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// correct with spacing required
const [user] = await sql `SELECT * FROM users WHERE id = ${id}`
ctx.body = {
  name: user.name,
  avatar: escape `/avatars/${user.id}.jpg`,
}

// correct with spacing disallowed
const [user] = await sql`SELECT * FROM users WHERE id = ${id}`
ctx.body = {
  name: user.name,
  avatar: escape`/avatars/${user.id}.jpg`,
}
```

**Comment by platinumazure** [comment]
Seems reasonable to me-- I'll :+1: this.

**Comment by vitorbal** [comment]
I'll :+1: this too. I think we should allow configuring "always" or "never"
having spacing btwn them.

**Comment by isiahmeadows** [comment]
:+1: from me, too. And I agree @vitorbal (although one space should also be an option).

**Comment by vitorbal** [comment]
@phaux are you willing to work on a PR for this if it gets accepted?

**Comment by phaux** [comment]
@vitorbal I think I can do it

**Comment by ilyavolodin** [comment]
Anyone on the team wants to champion this? It looks like there are more then enough votes to accept this, just need a champion.

**Comment by kaicataldo** [comment]
I'll champion this. @phaux Is everything clear? Please let us know if you need any help with anything, either here in this issue or in our [Gitter](https://gitter.im/eslint/eslint).

**Comment by vitorbal** [comment]
marking this as accepted as we have a champion and enough +1's

**Comment by jwilsson** [comment]
I'd like to take a stab at this if no one else is working on it, @phaux?

**Comment by kaicataldo** [comment]
@jwilsson Go for it! It looks like we didn't discuss a configuration schema, but it seems to me like this should have one string option that is either "always" or "never". Thoughts on what the default should be?

Aside from needing to figure out the default value, is everything else clear? Please feel free to comment on here or swing by the [ESLint Gitter](https://gitter.im/eslint/eslint) if you need any help!

**Comment by not-an-aardvark** [comment]
`"always"` and `"never"` sound good to me.

I think the default should be `"never"`, since I've never seen anyone intentionally use a space here.

**Comment by vitorbal** [comment]
Agreed on "never" by default

**Comment by kaicataldo** [comment]
Agreed :+1:

**Comment by jwilsson** [comment]
Cool, I'll take a look in the coming days.

**Comment by phaux** [comment]
I haven't began working on it yet so go ahead @jwilsson .

I agree upon the default, because it makes this new syntax stand out more. Putting a space there makes it look less like an invocation (even tho I think it looks prettier)
